modopt/opt/cost.py

```diff
@@ -183,7 +185,8 @@ def get_cost(self, *args, **kwargs):
         """
 
         # Check if the cost should be calculated
-        if self._iteration % self._cost_interval:
+        if self._cost_interval is None or \
+                self._iteration % self._cost_interval:
```

Comment:
I don't understand how this works: does that mean that the cost gets computed everytime the number of iteration is not a multiple of the cost interval? that seems weird. @chaithyagr

Reply:
The check is to see when the cost needs to be calculated, when the result of check is false, we calculate the cost function, else, we just return False.
So say cost_interval is 5, and there are 100 iterations, we calculate cost at every 5, 10, 15, 20 cost and so on.

Reply:
If the above statement is True, we dont calculate cost, thereby what happens is quite opposite of what you say.

Reply:
Sure I wasn't doing the computation correctly, sorry